pkg/types/baremetal/validation/platform.go

```diff
@@ -338,24 +338,52 @@ func validateNetworkConfig(hosts []*baremetal.Host, fldPath *field.Path) (errors
 	return
 }
 
-// ensure that the bootMode field contains a valid value
-func validateBootMode(hosts []*baremetal.Host, fldPath *field.Path) (errors field.ErrorList) {
+// validateBMC validates all host BMC addresses and calls custom validate function
+func validateBMC(hosts []*baremetal.Host, fldPath *field.Path, validateFn func(host *baremetal.Host, b bmc.AccessDetails, idx int, fldPath *field.Path) field.ErrorList) field.ErrorList {
```

Comment:
Don't want to be a blocking comment, but maybe the setup could be slightly simplified/more readable by adding a specific validation rule for the access details (to ensure that everything it's fine) to be invoked before `validateBootMode` and `validateProvisioningNetworkDisabledSupported` (respectively, in `ValidateProvisioning` and `ValidatePlatform`).
At this point, each validate function could simply iterate over all the hosts (as it happens in other validation rules), fetch the access details (now assuming that no error will happen, so just an additional line of code), and apply its own validation logic

Reply:
This PR is working, however I was thinking the same thing that hurts readability and agree with you. I'll update to simpler version like you suggested.